src/internal.c

```diff
@@ -2589,6 +2589,28 @@ int InitSSL_Ctx(WOLFSSL_CTX* ctx, WOLFSSL_METHOD* method, void* heap)
             ctx->CBIORecvFrom = uIPRecvFrom;
         }
         #endif
+    #elif defined(ARDUINO)
+        #if !defined(ARDUINO_ARCH_ESP32) && !defined(ARDUINO_ARCH_ESP8266) && \
+            !defined(ARDUINO_ARCH_SAMD)  && !defined(ARDUINO_ARCH_NRF52)   && \
+            !defined(ARDUINO_ARCH_MBED)  && !defined(ARDUINO_ARCH_STM32)   && \
+            !defined(ARDUINO_SEEED_XIAO) && !defined(ARDUINO_TEENSY41)     && \
+            !defined(WIFI_AVAILABLE)     && !defined(ETHERNET_AVAILABLE)
+            /* Unless a known board is detected, assume there's no network */
+            #define WOLFSSL_NO_NETWORK
+        #else
+            ctx->CBIORecv = EmbedReceive;
```

Comment:
You are duplicating all the code below for platforms that do support network IO. 

I'd prefer to see something in settings.h that detects this and sets WOLFSSL_USER_IO in the cases where there is no network.

What is `WOLFSSL_NO_NETWORK` for anyways? That doesn't exist in our code base. Please just gate on existing WOLFSSL_USER_IO.

Reply:
> You are duplicating all the code below for platforms that do support network IO.

Yes. That was intentional. I purposely pulled out the Arduino-specific code to isolate it from defaults that might later change, even though it was currently a duplicate of the default. See update that uses default for Arduino CB.

> What is WOLFSSL_NO_NETWORK for anyways? 

That was going to be a macro for Arduino boards known to have no networking capabilities. I've removed it.